### ISSUE TITLE
Support STSAssumeRoleSessionCredentialsProvider

### ISSUE THREAD

**Issue by avernonquantify** [jcaddel/maven-s3-wagon#34]
Where the user credentials are defined within one account and a cross account role provides the permissions to write to the S3 bucket there is a need to use the STSAssumeRoleSessionCredentialsProvider.

This issue would relate to supporting ~/.aws/credentials.

**Comment by dukejansen** [comment]
Indeed, just using the same chained provider that AWS Java SDK uses under the covers would be very helpful here.

**Comment by lareeth** [comment]
This pull request https://github.com/jcaddel/maven-s3-wagon/pull/33 fixes this issue, however we need Jeff to accept

CC @jcaddel @jeffcaddel

**Comment by avernonquantify** [comment]
Any update?

**Comment by avernonquantify** [comment]
Nearly a year later... Any update?
